<launch>
  <arg name="file_location" default="/tmp/test.avi" />
  <arg name="do_timestamp" default="false"/>
  <arg name="audio_format" default="mp3"/>
  <arg name="channels" default="1"/>
  <arg name="sample_rate" default="44100"/>
  <arg name="sample_format" default="S16LE"/>

  <node name="usb_cam_node" pkg="usb_cam" type="usb_cam_node" />

  <include file="$(find audio_capture)/launch/capture.launch">
    <arg name="format" value="$(arg audio_format)" />
    <arg name="channels" value="$(arg channels)" />
    <arg name="sample_rate" value="$(arg sample_rate)" />
    <arg name="sample_format" value="$(arg sample_format)" />
  </include>

  <include file="$(find audio_video_recorder)/launch/audio_video_recorder.launch">
    <arg name="audio_topic" value="/audio/audio" />
    <arg name="image_topic" value="/usb_cam_node/image_raw" />
    <arg name="file_location" value="$(arg file_location)" />
    <arg name="do_timestamp" value="$(arg do_timestamp)" />
    <arg name="audio_format" value="$(arg audio_format)" />
    <arg name="channels" value="$(arg channels)" />
    <arg name="sample_rate" value="$(arg sample_rate)" />
    <arg name="sample_format" value="$(arg sample_format)" />
  </include>
</launch>
